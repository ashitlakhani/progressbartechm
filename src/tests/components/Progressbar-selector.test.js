import React from 'react';
import { shallow } from 'enzyme';

import ProgressbarSelector from '../../src/components/Progressbar-selector';
import barsArr from '../testdata/barsArr-fixtures';

describe('In Progressbar-selector test ', () => {
    let wrapper, noBarSelectedWrapper, selectPbSpy;

    beforeEach(() => {
        selectPbSpy = jest.fn();
        wrapper = shallow(<ProgressbarSelector hasSelectedPb={ barsArr[0].pbId } bars={ barsArr } selectPbById={ selectPbSpy } />);
        noBarSelectedWrapper = shallow(<ProgressbarSelector hasSelectedPb={ '' } bars={ barsArr } />);
    });

    it('should render prog bar selector', () => {
        expect(wrapper).toMatchSnapshot();
    });

});
