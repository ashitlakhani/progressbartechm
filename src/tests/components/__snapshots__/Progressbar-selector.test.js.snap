// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`In Progressbar-selector test  should render prog bar selector 1`] = `
<FormGroup
  tag="div"
>
  <Label
    className="text-brand"
    for="progressbAs-select0"
    tag="label"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    Select Progressbar
  </Label>
  <Input
    bsSize="lg"
    className="rounded-0 normal-bdcolor-boxshadow custom-select height-3rem appear-btn"
    id="progressbAs-select0"
    name="progressbar-select"
    onChange={[Function]}
    type="select"
  >
    <ProgOptions
      key="0"
      pbId="pb1"
      val={10}
    />
    <ProgOptions
      key="1"
      pbId="pb2"
      val={81}
    />
    <ProgOptions
      key="2"
      pbId="pb3"
      val={42}
    />
  </Input>
</FormGroup>
`;

exports[`In components/Progressbar-selector.js Test Selector:  should choose progressbar. 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "target": Object {
          "value": "pb1",
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`In components/Progressbar-selector.js Test Selector:  should render Progressbar-selector correctly. 1`] = `
<FormGroup
  tag="div"
>
  <Label
    className="text-brand"
    for="progressbAs-select0"
    tag="label"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    Select Progressbar
  </Label>
  <Input
    bsSize="lg"
    className="rounded-0 normal-bdcolor-boxshadow custom-select height-3rem appear-btn"
    id="progressbAs-select0"
    name="progressbar-select"
    onChange={[Function]}
    type="select"
  >
    <option
      disabled={true}
      value="default"
    >
      Please Select
    </option>
    <PbOpts
      key="0"
      pbId="pb1"
      val={10}
    />
    <PbOpts
      key="1"
      pbId="pb2"
      val={81}
    />
    <PbOpts
      key="2"
      pbId="pb3"
      val={42}
    />
  </Input>
</FormGroup>
`;
